x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -373,6 +379,7 @@ export class RulesClient {
       updatedBy: username,
       createdAt: new Date(createTime).toISOString(),
       updatedAt: new Date(createTime).toISOString(),
+      snoozeEndTime: data.snoozeEndTime ? new Date(data.snoozeEndTime).toISOString() : undefined,
```

Comment:
Probably want to use `null` here instead of `undefined`.  By the time this makes it to the SO, we'll need to be storing the value as `null` and not `undefined`, as `undefined` means partial update for some of our update functionality.  Quite possible this all does the right thing once it gets deeper than the rulesClient (when it actually updates the SO), but ... seems safer to not be using `undefined` at all, in case something changes in the future.